public/app/features/panel/PanelRenderer.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useMemo } from 'react';
+import { applyFieldOverrides, FieldConfigSource, getTimeZone, PanelData, PanelPlugin } from '@grafana/data';
+import { PanelRendererProps } from '@grafana/runtime';
+import { config } from 'app/core/config';
+import { appEvents } from 'app/core/core';
+import { useAsync } from 'react-use';
+import { getPanelOptionsWithDefaults, OptionDefaults } from '../dashboard/state/getPanelOptionsWithDefaults';
+import { importPanelPlugin } from '../plugins/plugin_loader';
+
+export function PanelRenderer<P extends object = any, F extends object = any>(props: PanelRendererProps<P, F>) {
+  const {
+    pluginId,
+    data,
+    timeZone = getTimeZone(),
+    options = {},
+    width,
+    height,
+    title,
+    onOptionsChange,
+    onChangeTimeRange = () => {},
+    fieldConfig: config = { defaults: {}, overrides: [] },
+  } = props;
+
+  const [fieldConfig, setFieldConfig] = useState<FieldConfigSource>(config);
+  const { value: plugin, error, loading } = useAsync(() => importPanelPlugin(pluginId), [pluginId]);
+  const defaultOptions = useOptionDefaults(plugin, options, fieldConfig);
+  const dataWithOverrides = useFieldOverrides(plugin, defaultOptions, data, timeZone);
+
+  if (error) {
+    return <div>Failed to load plugin: {error.message}</div>;
+  }
+
+  if (loading) {
+    return <div>Loading plugin panel...</div>;
+  }
+
+  if (!plugin || !plugin.panel) {
+    return <div>Seems like the plugin you are trying to load does not have a panel component.</div>;
+  }
+
+  if (!dataWithOverrides) {
+    return <div>No panel data</div>;
+  }
+
+  const PanelComponent = plugin.panel;
+
+  return (
+    <PanelComponent
```

Comment:
I think this is a good idea 👍

Reply:
Yea, an error boundary here does make sense

Reply:
hmm, now when I play around with it I might change my mind. Since the PanelChrome is outside of this component and we probably want to mark that component when an error occurs we should let the consumer handle this instead of doing it for them.